### DISCUSSION TITLE
How to update `solid-start`?

### DISCUSSION THREAD

**Discussion by PeronGH** [solidjs/solid-start#1350] [Q&A] [answered]
How to migrate from old versions to the latest version of `solid-start`? Is there any migration tools or instructions?

**Comment by atilafassina** [comment, marked as answer]
There isn't a particular migration guide.

At this point, there are 2 breaking-changes you need to be aware of.

- Solid-Router 0.9 to 0.10
- SolidStart

The thing is, the data loading methods that were present in SolidStart are gone and favor of the new ones that were added in Solid-Router. You can get around with `createResource` on your first step migration. So, I wouldn't recommend you take too long before updating Solid-Router. Though it is possible to do them as 2 separate steps.

- [ ] Replace every method with `$` in SolidStart with something present in Solid-Router or Solid core.
- [ ] Replace `vite.config.js` with `app.config.js` (this is simply inverts some declarations, as in: `start:{}` is root-level and Vite configs are `vite:{}`. So, using `defineConfig` from SolidStart would assist in doing it if you're in a TS environment.

When migrating to current Solid-Router you should:
- [ ] Replace `createResource` with `createAsync` and probably wrap them in a `cache()` (both require Solid-Router 0.10)
- [ ] Drop every `<Outlet>` and use `props.children` instead.

> [!Warning]
> if you're using `<Routes>`, you'll have to replace it too and that's probably the more nuanced context-based issue. Often you can get by with nesting `<Router>` declarations.
